security/security-secrets-setup.1.rst

```diff
@@ -0,0 +1,191 @@
+NAME
+====
+
+security-secrets-setup — Creates an on-device public-key infrastructure (PKI) to secure microservice secret management
+
+SYNOPSIS
+========
+
+| security-secrets-setup [-confdir confdir] -config params.json
+| security-secrets-setup [-confdir confdir] -generate
+| security-secrets-setup [-confdir confdir] -cache
+| security-secrets-setup [-confdir confdir] -import
+| security-secrets-setup [-h|--help]
+
+
+DESCRIPTION
+===========
+
+The Vault secret management component of EdgeX Foundry requires TLS encryption
+of secrets over the wire via a pre-created PKI.
+security-secrets-setup is responsible for creating a certificate authority
+and any needed TLS leaf certificates in order to secure the EdgeX security services.
+security-secrets-setup supports several modes of operation as defined in the OPTIONS section.
+
+As the PKI is security-sensitive,
+this tool takes a number of precautions to safeguard the PKI:
+
+* The PKI can be deployed to transient storage to address potential attacks to the PKI at-rest.
+
+* The PKI is deployed such that each service has its own assets folder, which is amenable to security controls imposed by container runtimes such as mandatory access controls or file system namespaces.
+
+* The private key of the certificate authority (CA) is shredded (securely erased) prior to caching or deployment to block issuance of new CA descendants (this is most relevant in caching mode).
+
+
+OPTIONS
+=======
+
+Modes of operation
+------------------
+
+  \-config *params.json*
+    Legacy option from prior version of security-secrets-setup.
+    Deploys PKI using existing behavior in Edinburgh release.
+
+  \-generate
+    Causes a PKI to be generated afresh every time and deployed.
+    Typically, this will be whenever the framework is started.
+
+  \-cache
+    Causes a PKI to be generated exactly once and then copied
+    to a designated cache location for future use.
+    The PKI is then deployed from the cached location.
+
+  \-import
+    This option is similar to ``-cache`` in that it deploys a PKI
+    from *CacheDir* to *DeployDir*, but it forces an error if
+    *CacheDir* is empty instead of triggering PKI generation.
+    This enables usage models for deploying a pre-populated PKI
+    such as a Kong certificate signed by an external certificate authority
+    or TLS keys signed by an offline enterprise certificate authority.
+
+  \-h
+    Display help text
+
+  \-help
+    Display help text
+
+  \-confdir confdir
```

Comment:
what was the result of your investigation into the other edgex-go services' options for the profile, confdir, etc.?

Reply:
-optname and --optname seem to be handled identically.  -confdir is provided by pkg/config/loader.go and -p/-profile are reimplemented by utilities.  Forgot to fix this section.

Reply:
`-confdir` is also used on other edgex-go services as well I believe, not just here.  -profile is also used in the whole edgex-go widely too. so this is not something newly introduced here.

Reply:
Also current utility doesn't support profiles nor can it talk to the registry since that service doesn't exist at the time this tool runs.

Reply:
This section has been fixed up, I hope.

Reply:
AFAIK, profile has nothing to do with consul that's just to switch between `res/configuration.toml` and `res/docker/configuration.toml`

Reply:
yeap, that is most used in `-profile` currently and `-confdir` allows you to choose different resource directory other than the default `./res`

Reply:
So... to turn this into something actionable.  does the current utility implement -profile.  If not, is there a call to add it?  I will update the docs to reflect whatever is decided.

Reply:
Yes I'd say this is a call to add it

Reply:
I just checked `main.go` and it is already there in the code ported by TingYu.  So this will be a doc-only update.

Reply:
Will update docs now.  Will add note to the user story to add support for the switch when adding support for configuration.toml in a later PR.